### PR TITLE
Add OD Documentation

### PR DESCRIPTION
**Pull request by liopeer**
## What has changed and why?
- adds docs on how to do OD training
- moves training, loading and predict functions to the public API

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Will be part of the big new update anyway

## PDF (Screenshots)
[od_page.pdf](https://github.com/user-attachments/files/23191564/od_page.pdf)
[api_page.pdf](https://github.com/user-attachments/files/23191584/api_page.pdf)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ class _DINOv3LTDETRObjectDetectionTinyConfig(_DINOv3LTDETRObjectDetectionConfig)
     )
 
 
-class DINOv3LTDETRObjectDetectionTaskModel(TaskModel):
+class DINOv3LTDETRObjectDetection(TaskModel):
```
Comment (liopeer) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py:208]: Unrelated, but should be fixed. Will also fix this in the checkpoints.

File: docs/source/object_detection.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+After the training completes, you can load the best model checkpoints for inference like this:
+
+```python
+import lightly_train
+
+model = lightly_train.load_model_from_checkpoint(
+    "out/my_experiment/exported_models/exported_best.pt"
+)
+masks = model.predict("path/to/image.jpg")
```
Comment (stegmuel) [inline docs/source/object_detection.md]:
```suggestion
results = model.predict("path/to/image.jpg")
```

File: docs/source/object_detection.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+Or use one of the pre-trained model weights directly from LightlyTrain:
+
+```python
+import lightly_train
+
+model = lightly_train.load_model_from_checkpoint(
+    checkpoint="dinov3/convnext-tiny-ltdetr-coco"
+)
+masks = model.predict("path/to/image.jpg")
```
Comment (stegmuel) [inline docs/source/object_detection.md]:
```suggestion
results = model.predict("path/to/image.jpg")
```

File: docs/source/object_detection.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        checkpoint="out/my_experiment/exported_models/exported_best.pt", # use the best model to continue training
+        data={...},
+    )
+```
+
+<!-- TODO (Lionel, 10/25) Add instructions for re-using classification head when it is supported. -->
+
+(object-detection-use-model-weights)=
+
+### Load the Trained Model from Checkpoint and Predict
```
Comment (stegmuel) [inline docs/source/object_detection.md:79]: This section seems to only bring information that is already contained in the "Visualize" section. Should it be merged?

File: docs/source/object_detection.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+)
+masks = model.predict("path/to/image.jpg")
+```
+
+Or use one of the pre-trained model weights directly from LightlyTrain:
+
+```python
+import lightly_train
+
+model = lightly_train.load_model_from_checkpoint(
```
Comment (stegmuel) [inline docs/source/object_detection.md:97]: We would still need to add support for directly (down)loading our checkpoints.

File: docs/source/object_detection.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    boxes=boxes,
+    labels=[model.classes[i.item()] for i in labels],
+)
+
+fig, ax = plt.subplots(figsize=(30, 30))
+ax.imshow(image_with_boxes.permute(1, 2, 0))
+fig.savefig("predictions.png")
+```
+
+The predicted boxes are in the normalized (x_min, y_min, x_max, y_max) format.
```
Comment (stegmuel) [inline docs/source/object_detection.md]: I don't think they are normalized.

File: docs/source/object_detection.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+Create a new output directory for each experiment to keep training logs, model exports,
+and checkpoints organized.
+```
+
+(object-detection-data)=
+
+## Data
+
+Lightly**Train** supports training object detection models with images and bounding boxes.
+Every image must have a corresponding annotation file (in [YOLO format](https://labelformat.com/formats/object-detection/yolov5/)) that contains on every line 4 normalized bounding box coordinates (x_center, y_center, width, height) and class IDs for each object in the image.
```
Comment (stegmuel) [inline docs/source/object_detection.md]: Maybe add an example of annotation file.
I would also mention the class ID first as it comes first in the line.

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 8]
```diff
@@ -23,7 +23,7 @@
 from lightly_train._distributed import reduce_dict
 from lightly_train._task_checkpoint import TaskSaveCheckpointArgs
 from lightly_train._task_models.dinov3_ltdetr_object_detection.task_model import (
-    DINOv3LTDETRObjectDetectionTaskModel,
+    DINOv3LTDETRObjectDetection,
```
Comment (stegmuel) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:26]: You can modify this, but I think there are more places where this is wrong and I'm doing it anyway.

File: docs/source/object_detection.md (hunk 5 above)
Comment (liopeer) [reply docs/source/object_detection.md:97]: That should have been done in https://github.com/lightly-ai/lightly-train/pull/373

File: docs/source/object_detection.md (hunk 6 above)
Comment (liopeer) [reply docs/source/object_detection.md]: You are completely right!

Comment (liopeer) [reply docs/source/object_detection.md]:
```suggestion
The predicted boxes are in the absolute (x_min, y_min, x_max, y_max) format, i.e. represent the size of the dimension of the bounding boxes in pixels.
```

File: docs/source/object_detection.md (hunk 4 above)
Comment (liopeer) [reply docs/source/object_detection.md:79]: It's a bit redundant I agree, but I thought it would be good to have both examples, i.e. loading from local file and loading from a LightlyTrain included checkpoint.

Comment (liopeer) [reply docs/source/object_detection.md:79]: I would leave it and then re-do it when we make the docs anyway more fine-tuning centric. I would then make some high-level explanation that covers some functionality that is shared between sem seg and OD (and all the others).

File: docs/source/object_detection.md (hunk 7 above)
Comment (liopeer) [reply docs/source/object_detection.md]: Good idea. Updated!

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py (hunk 8 above)
Comment (liopeer) [reply src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:26]: The linter will tell us if we miss it somewhere, so I'd keep it.
